### PR TITLE
CRM-18652 - Show 'no matches' if quick search returns no results

### PR DESCRIPTION
**Pull request by ineffyble**
---
- CRM-18652: Quick search should provide feedback to user if no matches are found
  https://issues.civicrm.org/jira/browse/CRM-18652

### CODE REVIEW

**Comment by kurund** [comment]
jenkins, test this please

File: templates/CRM/common/navigation.js.tpl [hunk 1]
```diff
@@ -103,10 +103,12 @@ $('#civicrm-menu').ready(function() {
...
         CRM.api3('contact', 'getquick', params).done(function(result) {
           var ret = [];
-          if (result.values) {
+          if (result.values.length > 0) {
             $.each(result.values, function(k, v) {
               ret.push({value: v.id, label: v.data});
-            })
+            });
+          } else {
+            ret.push({value: '0', label: 'No matches'});
```
Comment (kurund) [inline templates/CRM/common/navigation.js.tpl]: label needs to be localized.

Comment (ineffyble) [inline templates/CRM/common/navigation.js.tpl]: Good point - fixed now.

**Comment by totten** [comment]
jenkins, add to whitelist

**Comment by alifrumin** [comment]
@josephlacey and I are looking at this one

**Comment by alifrumin** [comment]
HI @ineffyble (and @seamuslee001),

Tested this on 4.7.9 and the new feature works great.  If you enter text into the Quicksearch and no matches are found, the dropdown shows "None Found".

However if you press enter with text in the search box  and there are no matching contacts, you get an error that reads:  "We could not find a contact id."  This is a regression because the current behaviour should send the unfound results to the Advanced Search.

We believe this has something to do with that the value being set to 0, so Civi is trying to load contact with id 0.  This error should be addressed before we merge this.

**Comment by alifrumin** [comment]
@seamuslee001 is gonna take a look

**Comment by alifrumin** [comment]
@eileenmcnaughton I believe this issue can be closed as https://github.com/civicrm/civicrm-core/issues/8495 addressed it and has been merged

**Comment by eileenmcnaughton** [comment]
Thanks Alice - closing

**Pull request closed without merging**